# HG changeset patch
# User oleh-derevenko
# Date 1547341122 -7200
# Node ID ea98ff157e843bdee7c7c4eee773bdf04d274bae
# Parent  3262016ef92b4925253411592270c10899e9c652
Fixed: Some more architectures support added for Debian builds

diff --git a/include/ode/odeconfig.h b/include/ode/odeconfig.h
--- a/include/ode/odeconfig.h
+++ b/include/ode/odeconfig.h
@@ -80,8 +80,10 @@
 #endif // #if !defined(__GNUC__)
 
 
-/* Well-defined common data types...need to define for 64 bit systems */
-#if defined(__aarch64__)
+/* Well-defined common data types...need to be defined for 64 bit systems */
+#if defined(__aarch64__) || defined(__alpha__) || defined(__ppc64__) \
+    || defined(__s390__) || defined(__s390x__) || defined(__zarch__) \
+    || defined(__mips__) || defined(__powerpc64__) || defined(__riscv) || defined(__sparc64__) || defined(__sparc__)
     #include <stdint.h>
     typedef int64_t         dint64;
     typedef uint64_t        duint64;
@@ -97,13 +99,13 @@
     typedef ptrdiff_t       ddiffint;
     typedef size_t          dsizeint;
 
-#elif defined(_M_IA64) || defined(__ia64__) || defined(_M_AMD64) || defined(__x86_64__)
+#elif (defined(_M_IA64) || defined(__ia64__) || defined(_M_AMD64) || defined(__x86_64__)) && !defined(__ILP32__) && !defined(_ILP32)
   #define X86_64_SYSTEM   1
 #if defined(_MSC_VER)
   typedef __int64         dint64;
   typedef unsigned __int64 duint64;
 #else
-#if defined(_LP64)
+#if defined(_LP64) || defined(__LP64__)
 typedef long              dint64;
 typedef unsigned long     duint64;
 #else
